### PR TITLE
audit: Silence not notable on non-Homebrew taps

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Silence the `brew audit` error message `GitHub repository not notable` for non-Homebrew taps.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -590,7 +590,8 @@ def audit_github_repository
     return if metadata.nil?
 
     problem "GitHub fork (not canonical repository)" if metadata["fork"]
-    if (metadata["forks_count"] < 20) && (metadata["subscribers_count"] < 20) &&
+    if user.casecmp("homebrew").zero? &&
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]: Use `formula&.tap&.core_tap?`

Comment (sjackman) [reply Library/Homebrew/dev-cmd/audit.rb]: Done.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Comment by sjackman** [comment]
Thanks for merging, Mike!

**Pull request merged**